### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by lriggs**
Example of adding two arbitrary text boxes to the connection dialog and using the values in the connection builder.

### CODE REVIEW

File: samples/components/dialogs/new text field/connection-dialog.tcd [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<connection-dialog class='kyvos'>
```
Comment (pvanderknyff) [inline samples/components/dialogs/new text field/connection-dialog.tcd]: Here, the class is 'kyvos', and below, in the tdr it's 'postgres_odbc2'. Even if this isn't a full plugin (I don't see a manifest), they should probably be aligned. Maybe use 'new_text_field_sample' to be descriptive?

Comment (lriggs) [reply samples/components/dialogs/new text field/connection-dialog.tcd]: Thanks good catch.

Comment (lriggs) [reply samples/components/dialogs/new text field/connection-dialog.tcd]: Interestingly it worked like this.

**Comment by lriggs** [comment]
> 
> 
> Just for clarity's sake on my part, this sample using existing text boxes in new ways, correct?

Yes.

**Comment by rosswbrown** [comment]
Service is a great example of an arbitrary extension point in the dialog.  Warehouse gets a little trickier longer term.  We likely in the future will want to allow HasWarehouses to be true in the tcd as a part of more flexible metadata enumeration, potentially giving that attribute meaning more globally in the platform.  No objections though.

File: samples/components/dialogs/new text field/connectionRequired.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+(function requiredAttrs(attr)
+{
+    return ["class", "server", "port", "dbname", "username", "password", connectionHelper.attributeWarehouse, "service"];
```
Comment (rosswbrown) [inline samples/components/dialogs/new text field/connectionRequired.js]: more consistent usage of connectionHelper in samples when possible

**Pull request merged**